src/test/regress/sql/multi_name_resolution.sql

```diff
@@ -0,0 +1,27 @@
+--
+-- MULTI_NAME_RESOLUTION
+--
+
+-- There was a failure compiling queries with shadowed subquery aliases
+-- https://github.com/citusdata/citus/issues/2642
+
+create table namenest1 (id integer primary key, user_id integer);
+create table namenest2 (id integer primary key, value_2 integer);
+select * from create_distributed_table('namenest1', 'id');
+select * from create_reference_table('namenest2');
+
+SELECT r
+FROM (
+	SELECT user_id_deep, random() as r -- prevent pulling up the subquery
+	FROM (
+		namenest1
+		JOIN namenest2 ON (namenest1.user_id = namenest2.value_2)
+	) AS join_alias(user_id_deep)
+) AS bar, (
+	namenest1
+	JOIN namenest2 ON (namenest1.user_id = namenest2.value_2)
+) AS join_alias(user_id_deep)
+WHERE (bar.user_id_deep = join_alias.user_id_deep);
+
+drop table namenest1;
+drop table namenest2;
```

Comment:
for cleanliness and to prevent naming conflicts for tests that run concurrently, we often do something like:

```sql
CREATE SCHEMA multi_name_resolution;
SET search_path TO multi_name_resolution;
...
DROP SCHEMA multi_name_resolution CASCADE;
```

Reply:
Updated